map_view/display-device-location/src/main/java/com/esri/samples/display_device_location/DisplayDeviceLocationSample.java

```diff
@@ -0,0 +1,197 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.display_device_location;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Calendar;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.CornerRadii;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.geometry.Geometry;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.location.SimulatedLocationDataSource;
+import com.esri.arcgisruntime.location.SimulationParameters;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.LocationDisplay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+import org.apache.commons.io.IOUtils;
+
+public class DisplayDeviceLocationSample extends Application {
+
+  private MapView mapView;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Display Device Location Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+      scene.getStylesheets().add(getClass().getResource("/display_device_location/style.css").toExternalForm());
+
+      // create ArcGISMap with the imagery basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create combo box
+      ComboBox<String> comboBox = new ComboBox<>();
+      comboBox.setMaxWidth(Double.MAX_VALUE);
+      comboBox.setDisable(true);
+      // add the autopan modes to the combo box
+      comboBox.getItems().addAll("Off", "Recenter", "Navigation", "Compass");
```

Comment:
I'm a bit puzzled why in the app, when setting this to compass, the movement seems to switch to a more step-by-step style rather than a fluid movement of the position. Is this on purpose?